qiita_db/analysis.py

```diff
@@ -241,9 +241,11 @@ def dropped_samples(self):
         for biom, filepath in viewitems(bioms):
             table = load_table(filepath)
             # remove the samples from the sets as they are found in the table
-            proc_data_id = table.metadata()[0]['Processed_id']
+            proc_data_ids = set(sample['Processed_id']
+                                for sample in table.metadata())
             ids = set(table.ids())
-            all_samples[proc_data_id] = all_samples[proc_data_id] - ids
+            for proc_data_id in proc_data_ids:
```

Comment:
OK, I can't do this sanely or quickly. I need to add an entirely new study with all data to test this, AKA almost the entire populate_test_db.py needs to be duplicated for another study. This also means changing every test for everything else to reflect that new study existing. This seems like a huge time sink given the other more pressing things we need to do.